### PR TITLE
load_earth_age: Add inline examples

### PR DESCRIPTION
**Pull request by willschlitzer**
Add an inline example for `load_earth_age`.

Addresses #2225

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/datasets/earth_age.py [hunk 1]
```diff
@@ -57,6 +59,21 @@ def load_earth_age(resolution="01d", region=None, registration=None):
     The :class:`xarray.DataArray` grid doesn't support slice operation, for
     Earth seafloor crustal age with resolutions of 5 arc-minutes or higher,
     which are stored as smaller tiles.
+
+    Examples
+    --------
+
+    >>> from pygmt.datasets import load_earth_age
+    >>> # load the default grid (gridline-registered 01d grid)
+    >>> grid = load_earth_age()
```
Comment (yvonnefroehlich) [inline pygmt/datasets/earth_age.py:68]:
```suggestion
    >>> # load the default grid (gridline-registered 1 arc-degree grid)
    >>> grid = load_earth_age()
```

File: pygmt/datasets/earth_age.py [hunk 2]
```diff
@@ -57,6 +59,21 @@ def load_earth_age(resolution="01d", region=None, registration=None):
...
+    >>> # load the 30m grid with "gridline" registration
+    >>> grid = load_earth_age(resolution="30m", registration="gridline")
```
Comment (yvonnefroehlich) [inline pygmt/datasets/earth_age.py:70]: I feel for the sake of clarity it's  good to state the unit in long form.

```suggestion
    >>> # load the 30 arc-minute grid with "gridline" registration
    >>> grid = load_earth_age(resolution="30m", registration="gridline")
```

File: pygmt/datasets/earth_age.py [hunk 3]
```diff
@@ -57,6 +59,21 @@ def load_earth_age(resolution="01d", region=None, registration=None):
...
+    >>> # load high-resolution grid for a specific region
+    >>> grid = load_earth_age(
+    ...     resolution="05m",
+    ...     region=[120, 160, 30, 60],
+    ...     registration="gridline",
+    ... )
     """
```
Comment (yvonnefroehlich) [inline pygmt/datasets/earth_age.py:77]:
```suggestion
    >>> # load high-resolution (5 arc-minute) grid for a specific region
    >>> grid = load_earth_age(
    ...     resolution="05m",
    ...     region=[120, 160, 30, 60],
    ...     registration="gridline",
    ... )
    """
```

**Comment by yvonnefroehlich** [comment]
The following goes beyond this PR. At the moment I am wondering if the two messages blow are contradicting:

When running 
```
from pygmt.datasets import load_earth_age
grid = load_earth_age(resolution="30m")
```
It says:
```
pygmt-session [WARNING]: Remote dataset given to a data processing module but no registration was specified - 
default to gridline registration (if available)
```
In the [documentation]( https://www.pygmt.org/dev/api/generated/pygmt.datasets.load_earth_age.html#pygmt.datasets.load_earth_age) users find:
> registration (str) – Grid registration type. Either "pixel" for pixel registration or "gridline" for gridline registration. Default is None, where a pixel-registered grid is returned unless only the gridline-registered grid is available.

**Comment by seisman** [comment]
> > The following goes beyond this PR. At the moment I am wondering if the two messages blow are contradicting:
> > When running
> > ```
> > from pygmt.datasets import load_earth_age
> > grid = load_earth_age(resolution="30m")
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > It says:
> > ```
> > pygmt-session [WARNING]: Remote dataset given to a data processing module but no registration was specified - 
> > default to gridline registration (if available)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > In the [documentation](https://www.pygmt.org/dev/api/generated/pygmt.datasets.load_earth_age.html#pygmt.datasets.load_earth_age) users find:
> > > registration (str) – Grid registration type. Either "pixel" for pixel registration or "gridline" for gridline registration. Default is None, where a pixel-registered grid is returned unless only the gridline-registered grid is available.
> 
> As I understand it, this is coming from the default registration in GMT. Since an argument isn't passed to `registration` in this example, the API returns this warning.

See https://github.com/GenericMappingTools/pygmt/issues/1929 for the related issue report. We need to handle this in a separate PR.

**Review by michaelgrund — APPROVED** [review]

**Comment by yvonnefroehlich** [comment]
> > > The following goes beyond this PR. At the moment I am wondering if the two messages blow are contradicting:
> > > When running
> > > ```
> > > from pygmt.datasets import load_earth_age
> > > grid = load_earth_age(resolution="30m")
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > It says:
> > > ```
> > > pygmt-session [WARNING]: Remote dataset given to a data processing module but no registration was specified - 
> > > default to gridline registration (if available)
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > In the [documentation](https://www.pygmt.org/dev/api/generated/pygmt.datasets.load_earth_age.html#pygmt.datasets.load_earth_age) users find:
> > > > registration (str) – Grid registration type. Either "pixel" for pixel registration or "gridline" for gridline registration. Default is None, where a pixel-registered grid is returned unless only the gridline-registered grid is available.
> > 
> > 
> > As I understand it, this is coming from the default registration in GMT. Since an argument isn't passed to `registration` in this example, the API returns this warning.
> 
> See #1929 for the related issue report. We need to handle this in a separate PR.

Thanks @seisman for pointing out the related issue!

**Pull request merged**